custom_components/spotcast/helpers.py

```diff
@@ -103,117 +107,324 @@ async def run(*args, loop=None, executor=None, **kwargs):
+    searchType = "artist"
+    search = searchType + ":" + artistName
 
-    results = []
+    artistUri = ""
 
-    for searchType in SEARCH_TYPES:
-
-        try:
+    # get artist uri
+    try:
 
-            result = spotify_client.search(
-                searchType + ":" + search,
-                limit=1,
-                offset=0,
-                type=searchType,
-                market=country)[searchType + 's']['items'][0]
-
-            results.append(
-                {
-                    'name': result['name'].upper(),
-                    'uri': result['uri']
-                }
+        artist = spotify_client.search(
+            q=search,
+            limit=1,
+            offset=0,
+            type="artist",
+            market=country,
+        )["artists"]["items"][0]
+
+        _LOGGER.debug("found artist %s: %s", artist["name"], artist["uri"])
+        artistUri = artist["uri"]
+
+    except IndexError:
+        pass
+
+    results = spotify_client.artist_top_tracks(artistUri)
+    for track in results["tracks"][:10]:
+        _LOGGER.debug("track    : " + track["name"])
+
+    return results["tracks"]
+
+
+def get_search_string(
+    artistName: str,
+    albumName: str,
+    trackName: str,
+    genreName: str,
+    playlistName: str,
+    showName: str,
+    episodeName: str,
+    audiobookName: str,
+) -> str:
+    search = []
+    if not is_empty_str(artistName):
+        search.append(f"artist:{artistName}")
+        search.append(artistName)
+    if not is_empty_str(albumName):
+        search.append(f"album:{albumName}")
+        search.append(albumName)
+    if not is_empty_str(trackName):
+        search.append(f"track:{trackName}")
+        search.append(trackName)
+    if not is_empty_str(genreName):
+        search.append(f"genre:{genreName}")
+        search.append(genreName)
+    # if we are searching for a playlist, podcast, audiobook, we need some search query which is probably just the text we are looking for
+    for item in [playlistName, showName, episodeName, audiobookName]:
+        if not is_empty_str(item):
+            search.append(item)
+
+    return " ".join(search)
+
+
+# "album", "artist", "playlist", "track", "show", "episode", "audiobook"
+def get_types_string(
+    artistName: str,
+    albumName: str,
+    trackName: str,
+    playlistName: str,
+    showName: str,
+    episodeName: str,
+    audiobookName: str,
+) -> str:
+    types = []
+    if not is_empty_str(artistName):
+        types.append("artist")
+    if not is_empty_str(albumName):
+        types.append("album")
+    if not is_empty_str(trackName):
+        types.append("track")
+    if not is_empty_str(playlistName):
+        types.append("playlist")
+    if not is_empty_str(showName):
+        types.append("show")
+    if not is_empty_str(episodeName):
+        types.append("episode")
+    if not is_empty_str(audiobookName):
+        types.append("audiobook")
+    return ",".join(types)
```

Comment:
we can query for multiple types. Probably the "right" thing to do is create some more complex tree logic like, "if artist and song is set, then set type = song", since probably we are looking for a song.